### DISCUSSION TITLE
Balancing and ideas

### DISCUSSION THREAD

**Discussion by BreadDain** [IHateMyKite/UnforgivingDevices#50] [Ideas]
Decided to scribble yet another discussion concerning mod's balance. It is not a primary development focus nor it is anything related to issues, so thought here would be a proper place.

- Cuffs tying mechanics is really great, but it becomes too much of annoyance soon enough if represented as a fixed timed mechanic, can't wear them just for fun since they hinder/paralyze player a lot. This will come as a deterrent for lots of users, so probably is wise to reconsider it as either a modifier or a triggered effect without a timer, like controllable plug vibrations for example.
- Again related to cuffs: untying has quite a harsh difficulty, it is usually easier to just get rid of device alltogether without untying, if possible.
- (Idea) 2 stages of leg tying severity, same as 2 types of dresses, and 3 stages for arms - elbow hook, armbinder and elbowbinder.
- Any type of gag that contains "Extreme" in its name are the hardest devices possible in the mod, surpassing even elbowbinders sometimes. I think they shouldn't be harder than regular harness gags.
- Key minigame, the topic was brought up on LL thread. It is should be easy enough to unlock, but can contain a lot of other conditions or tricks. Since there is no exhaustion debuff inflicted during this minigame, it would probably be good if the minigame won't stop after one of locks was unlocked, so the player can unlock the whole device in one go. On other hand, device can be really mean and chew down the key on Sentient trigger, destroying it and jamming the lock, or trigger the plug randomly directly with sudden vibrations causing the player to drop they key and lose it permanently, or additionally shut itself closed with a lock shield, preventing any lockpicking/key unlock attempts. So the minigame itself doesn't focus on how easy is the device to unlock but rather how easy is to lose they key if you are not careful.
- (Idea) Another thing on unlocking minigame - it can just be converted to entirely different type of minigame, sort of "Keep the meter at 50% and hit a special button" minigame, imitating player trying to put the key into the keyhole with random meter increases/decreases, or going left-right similar to the key traveling around keyhole. With magicka/stamina manipulating bar movement direction, special key is being a 'Focus' button that forces magicka and stamina to drain quickly but making it easier to make it to the keyhole, health button is "Force the key in", accessibility dictating how hard it is to get the minibar closer to the middle and how fast the bar is trying to move away from the middle point. With wrong Health button key press triggering the sentience.
- Addressing the issue where certain plugs could lock the player in an orgasm loop and bind them very tight without giving any space to escape - plugs/piercing can be just maintaining certain amount of items always equipped without discriminating exact type they equip. So MAO/MAH modifiers trigger as usual per their condition, but instead a cumulative chance and amount of modifiers is calculated, let's say 6 devices from all modifiers together, and if player doesn't have that much equipped beside those that provide said modifiers - equip amount between Random(1,2) and half of the missing, clamped to highest value. If amount of devices is more than maintained amount - do nothing. Example - player have piercings that supposed to maintain 3 devices equipped, on first MAO trigger they equip leg and arm cuffs, on the second - only collar, on the third they will do nothing. But if player decides to struggle from all cuffs and collar - on next MAO trigger they will equip 1-2 other devices, for example blindfold and gag.
The chance is calculated whether at least 1 device can trigger from all those that have a modifier. So let's say MAO chances are 30, 40 and 50 on 3 different devices, the final modifier will be `1-(0,7*0,6*0,5) = 0.79`, that at least one of them works on MAO, but it will cause only one event equipping one specific amount of devices. The calculation can be either cumulative or multiplicative. This will also resolve event spam issue, where the UD asks every device if they supposed to do anything on any trigger because the values can be stored on equip/unequip.

**Comment by IHateMyKite** [comment]
>     * Cuffs tying mechanics is really great, but it becomes too much of annoyance soon enough if represented as a fixed timed mechanic, can't wear them just for fun since they hinder/paralyze player a lot. This will come as a deterrent for lots of users, so probably is wise to reconsider it as either a modifier or a triggered effect without a timer, like controllable plug vibrations for example.

I reworked it, so the cuffs can only be cooldown activated if they are sentient. Otherwise they can still be activated by other sentient devices.

> 
>     * Again related to cuffs: untying has quite a harsh difficulty, it is usually easier to just get rid of device alltogether without untying, if possible.

Will thing about this. Maybe I will make the devices harder to escape when actor is tied.

**Comment by IHateMyKite** [comment]
>     * (Idea) 2 stages of leg tying severity, same as 2 types of dresses, and 3 stages for arms - elbow hook, armbinder and elbowbinder.

Cuffs are already too much complex. Adding this feature will make it absolute hell to debug in future.

>     * Any type of gag that contains "Extreme" in its name are the hardest devices possible in the mod, surpassing even elbowbinders sometimes. I think they shouldn't be harder than regular harness gags.

All devices which have "Extreme" in their name are patched to be much harder. I can make them a little easier, but I will keep the idea same.

>     * Key minigame, the topic was brought up on LL thread. It is should be easy enough to unlock, but can contain a lot of other conditions or tricks. Since there is no exhaustion debuff inflicted during this minigame, it would probably be good if the minigame won't stop after one of locks was unlocked, so the player can unlock the whole device in one go. On other hand, device can be really mean and chew down the key on Sentient trigger, destroying it and jamming the lock, or trigger the plug randomly directly with sudden vibrations causing the player to drop they key and lose it permanently, or additionally shut itself closed with a lock shield, preventing any lockpicking/key unlock attempts. So the minigame itself doesn't focus on how easy is the device to unlock but rather how easy is to lose they key if you are not careful.
 
I was thinking early about adding some kind of modifier which makes the locks to lock themself once in the while to make them harder. On other hand the idea with lock shield is also nice. Perhaps it would be new minigame which player needs to finish before they can even acces the locks.
Idea with vibration making actor loose key is also great. I think it can be implemented quit easily. And it could work in 2 ways. When actor is in unlock minigame, and vibration start, it can cause actor to loose key. Or, as you said, the device itself can cause vibrations to start. But for it to work, I would have to rework the minigame as you said furthere. 

>     * (Idea) Another thing on unlocking minigame - it can just be converted to entirely different type of minigame, sort of "Keep the meter at 50% and hit a special button" minigame, imitating player trying to put the key into the keyhole with random meter increases/decreases, or going left-right similar to the key traveling around keyhole. With magicka/stamina manipulating bar movement direction, special key is being a 'Focus' button that forces magicka and stamina to drain quickly but making it easier to make it to the keyhole, health button is "Force the key in", accessibility dictating how hard it is to get the minibar closer to the middle and how fast the bar is trying to move away from the middle point. With wrong Health button key press triggering the sentience.

Nice idea, will think of how to implement it. I 

>     * Addressing the issue where certain plugs could lock the player in an orgasm loop and bind them very tight without giving any space to escape - plugs/piercing can be just maintaining certain amount of items always equipped without discriminating exact type they equip. So MAO/MAH modifiers trigger as usual per their condition, but instead a cumulative chance and amount of modifiers is calculated, let's say 6 devices from all modifiers together, and if player doesn't have that much equipped beside those that provide said modifiers - equip amount between Random(1,2) and half of the missing, clamped to highest value. If amount of devices is more than maintained amount - do nothing. Example - player have piercings that supposed to maintain 3 devices equipped, on first MAO trigger they equip leg and arm cuffs, on the second - only collar, on the third they will do nothing. But if player decides to struggle from all cuffs and collar - on next MAO trigger they will equip 1-2 other devices, for example blindfold and gag.

This is very good idea. Will give it a try, should be easy to implement. Amount of devices is already tracked by Register, so it should be simple.

>       The chance is calculated whether at least 1 device can trigger from all those that have a modifier. So let's say MAO chances are 30, 40 and 50 on 3 different devices, the final modifier will be `1-(0,7*0,6*0,5) = 0.79`, that at least one of them works on MAO, but it will cause only one event equipping one specific amount of devices. The calculation can be either cumulative or multiplicative. This will also resolve event spam issue, where the UD asks every device if they supposed to do anything on any trigger because the values can be stored on equip/unequip.

This sounds logical. But I don't think that this will have much of the impact. If you want, you can try to implement this feature yourself. Implement wise, it should be quite easy. You just have to make some actor check for all devices with MAO, and remove the check from OnOrgasm function. In other words, removing it from OnOrgasm and placing the modified version before OnOrgasm is called for all registered devices should do the trick.
